plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectPushdown.java

```diff
@@ -113,6 +115,31 @@ public static boolean isCompressionCodecSupported(InputFormat<?, ?> inputFormat,
         return false;
     }
 
+    public static boolean isSplittable(boolean s3SelectPushdownEnabled, Properties schema,
+                                       InputFormat<?, ?> inputFormat, Path path)
+    {
+        if (!s3SelectPushdownEnabled) {
+            return true;
+        }
+
+        if (isUncompressed(inputFormat, path)) {
+            Optional<S3SelectDataType> s3SelectDataTypeOptional = getDataType(getDeserializerClassName(schema));
+            return CSV.equals(s3SelectDataTypeOptional.orElse(null));
+        }
+
+        return false;
+    }
+
+    private static boolean isUncompressed(InputFormat<?, ?> inputFormat, Path path)
+    {
+        if (inputFormat instanceof TextInputFormat) {
```

Comment:
I used the same approach as in: https://github.com/trinodb/trino/blob/b97e7fa46f8120c8b94f8ea9d290a0c80372b1a9/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectPushdown.java#L104, for consistency, and I did not interfere with the code that did not concern my changes.
Do you want me to update that method as well with this pattern, since it is in the same class?

Reply:
Also, is this really a blocking change?

Reply:
No, this is rather a nit. Nothing blocking.

Reply:
Updated